transport-ble/src/main/java/com/ledger/live/ble/BleManager.kt

```diff
@@ -0,0 +1,418 @@
+                ScanSettings.CALLBACK_TYPE_FIRST_MATCH -> {
+                    Timber.d("Scan result => Match")
+                    val device = result.device
+                    val rssi = result.rssi
+
+                    if (scannedDevices.find { it.id == result.device.address } == null) {
+                        scannedDevices.add(
+                            BleDeviceModel(
+                                id = device.address,
+                                name = device.name,
+                                serviceId = device.uuids?.first()?.uuid.toString(),
+                                rssi = rssi
+                            )
+                        )
+
+                        onScanDevicesCallback?.invoke(scannedDevices)
+                    }
+                }
+                ScanSettings.CALLBACK_TYPE_MATCH_LOST -> {
+                    Timber.d("Scan result => Lost")
+                    if (scannedDevices.removeIf { it.id == result.device.address }) {
+                        onScanDevicesCallback?.invoke(scannedDevices)
+                    }
+                }
+            }
+            Timber.d("Scan Devices $scannedDevices")
+        }
+    }
+
+    var pollingJob: Job? = null
+    var onScanDevicesCallback: ((List<BleDeviceModel>) -> Unit)? = null
+
+    /**
+     * Use bleState for getting informations about running scan
+     */
+    fun startScanning(): Boolean {
+        return internalStartScanning()
+    }
+
+
+    fun startScanning(
+        onScanDevices: (List<BleDeviceModel>) -> Unit
+    ): Boolean {
+        Timber.d("Start Scanning")
+        onScanDevicesCallback = onScanDevices
+        return internalStartScanning()
+    }
+
+    /**
+     * Start a new scanning session
+     *
+     * Stop current device connection if exists
+     * Stop
+     */
+    private fun internalStartScanning(): Boolean {
+        //Assure to stop every runnning scan or active connection
+        disconnect()
+        stopScanning()
+
+        isScanning = true
+
+        val filters = mutableListOf<ScanFilter>()
+        //Filter NanoX service
+        filters.add(
+            ScanFilter.Builder()
+                .setServiceUuid(ParcelUuid(UUID.fromString(NANO_X_SERVICE_UUID)))
+                .build()
+        )
+
+        //Filter Faststack service
+        filters.add(
+            ScanFilter.Builder()
+                .setServiceUuid(ParcelUuid(UUID.fromString(NANO_FTS_SERVICE_UUID)))
+                .build()
+        )
+
+        scannedDevices = mutableListOf()
+
+        val scanSettings = ScanSettings.Builder()
+            .setScanMode(ScanSettings.SCAN_MODE_BALANCED)
+            .setCallbackType(ScanSettings.CALLBACK_TYPE_FIRST_MATCH)
+            .build()
+        bluetoothScanner.startScan(filters, scanSettings, scanCallback)
+
+        //Expose scanned device list every second
+        if (pollingJob == null) {
+            pollingJob = MainScope().launch(Dispatchers.IO) {
```

Comment:
[SHOULD] create a custom scope with a context (and even a SupervisorJob if you need to use the same scope with several couroutines: `val scope = CoroutineScope(Dispatchers.IO)`